### ISSUE TITLE
[FEATURE]: Support for Docker alternatives like nerdctl and Podman.

### ISSUE THREAD

**Issue by stefreak** [garden-io/garden#5014]
### Background / Motivation

Currently, when building containers locally, Garden invokes a `docker` binary that ships with Garden, and if Docker is not installed it crashes with an opaque error "Can't connect to the Docker server"

It might be that the user has Docker alternatives installed and available locally, like Podman and nerdctl. Those could be detected and used as a fallback implementation.

### What should the user be able to do?

Support nerdctl and Podman, and potentially other "Docker-compatible" container CLI implementations for Garden container builds in a local environment.

### Why do they want to do this? What problem does it solve?

Users are usually getting started with Garden by checking out one of our examples, which tend to default to the local environments. When trying out these examples, or when users need to use the local `local-kubernetes` provider for one reason or the other, we force them to install Docker.

### Suggested Implementation(s)

Detect that Docker is not installed, and fall back to using the `nerdctl` or `podman` CLIs from the local `PATH`.

### How important is this feature for you/your team?

🌵 Not having this feature makes using Garden painful

**Comment by worldofgeese** [comment]
Hi @stefreak, we have an ongoing issue for supporting this in issue #3137

**Comment by vvagaytsev** [comment]
@stefreak should we keep it as a separate issue or is it ok to close it in favor of #3137 as Tao suggested?

**Comment by stefreak** [comment]
Yes, can we migrate the technical details over e.g. the suggested implementation?

**Comment by vvagaytsev** [comment]
Sure, let's migrate all the necessary information.
